modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DeliverableAndesMetadata.java

```diff
@@ -482,6 +482,10 @@ private boolean isMarkAsAcked() {
             if (null != messageStatus && messageStatus.equals(ChannelMessageStatus.CLIENT_REJECTED)) {
                 continue;
             }
+            //the message could be recovered by one client and acknowledged by another, hence needs to be ignored
+            if (null != messageStatus && messageStatus.equals(ChannelMessageStatus.RECOVERED)) {
```

Comment:
Do we need to check null != messageStatus at each if? Can't we only check it at the top?